enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/EmbeddedInteraction.java

```diff
@@ -170,6 +171,13 @@ public void assertInitFailed( EnterpriseAuthSubject subject )
     }
 
     @Override
+    public void assertSessionKilled( EnterpriseAuthSubject subject )
+    {
+        assertThat( executeQuery( subject, "MATCH (n:Node) RETURN count(n)", null, r -> {} ),
+                containsString( "Read operations are not allowed" ) );
```

Comment:
Why is this parametrised with a subject? This assertion will give a false negative (as in: not throwing when it should) if the parameter subject doesn't have read privileges, right?